packages/runtime/types/locale.ts

```diff
@@ -0,0 +1,475 @@
+import { z } from 'zod'
+
+export const localeInputSchema = z.string().transform((value, ctx) => {
+  const { language, region } = new Intl.Locale(value)
+
+  if (
+    languageCodeSchema.safeParse(language as LanguageCode).success &&
+    region == null &&
+    value === language
+  ) {
+    return value as LocaleInput
```

Comment:
This casting shouldn't be necessary if we're using Zod and the parsed result

Reply:
But since `LocaleString` is just ``${CountryCode}-${LanguageCode}` | `${CountryCode}`` wouldn't that be inferred by TypeScript?

Reply:
Cool. Finally I could make it working by using
```ts
return `${languageResult.data}-${countryCodeResult.data}` as const
```